tests/jobs/nosim/test_nosim.py

```diff
@@ -0,0 +1,70 @@
+import os
+import shutil
+import subprocess
+import pytest
+import sys
+
+from ert_shared.plugins.plugin_manager import ErtPluginContext
+import semeio.hook_implementations.jobs
+import ert_shared.hook_implementations
+
+
+@pytest.mark.parametrize(
+    "nosim_command,data_input,data_expected",
+    [
+        ("INSERT_NOSIM", "RUNSPEC\n", "RUNSPEC\nNOSIM\n",),
+        (
+            "INSERT_NOSIM",
+            "RUNSPEC -- some comment about RUNSPEC and whatnot\n-- and more RUNSPEC\n",
+            "RUNSPEC\nNOSIM\n-- and more RUNSPEC\n",
+        ),
+        (
+            "INSERT_NOSIM",
+            "RUNSPEC some comment about RUNSPEC and whatnot\n-- and more RUNSPEC\n",
+            "RUNSPEC\nNOSIM\n-- and more RUNSPEC\n",
+        ),
+        ("REMOVE_NOSIM", "RUNSPEC\nNOSIM\n", "RUNSPEC\n",),
+        (
+            "REMOVE_NOSIM",
+            "RUNSPEC\nNOSIM\n-- and more RUNSPEC\n",
+            "RUNSPEC\n-- and more RUNSPEC\n",
+        ),
+        (
+            "REMOVE_NOSIM",
+            (
+                "RUNSPEC for some reason valid -- some comment RUNSPEC\n"
+                "-- more\n"
+                "NOSIM   more valid comment because it is after 8 chars"
+                " -- and now a valid comment\n"
+                "-- and a comment about NOSIM as well, not to be touched\n"
+            ),
+            (
+                "RUNSPEC for some reason valid -- some comment RUNSPEC\n"
+                "-- more\n-- and a comment about NOSIM as well, not to be touched\n"
+            ),
+        ),
+    ],
+)
+@pytest.mark.skipif(sys.version_info.major < 3, reason="requires python3")
+def test_nosim(setup_tmpdir, nosim_command, data_input, data_expected):
+    shutil.copy(os.path.join(os.path.dirname(__file__), "data", "nosim.ert"), ".")
+
+    with open("nosim.ert", "a") as fh:
+        fh.writelines(["FORWARD_MODEL {}".format(nosim_command)])
+
+    with open("TEST.DATA", "w") as fh:
+        fh.write(data_input)
+
+    with ErtPluginContext(
+        plugins=[semeio.hook_implementations.jobs, ert_shared.hook_implementations]
+    ):
+        p = subprocess.Popen(
+            ["ert", "test_run", "nosim.ert", "--verbose"],
+            stdout=subprocess.PIPE,
+            stderr=subprocess.PIPE,
+            env=os.environ.copy(),
+        )
+        p.wait()
+
+    with open("nosim/real_0/iter_0/TEST.DATA") as fh:
```

Comment:
I can´t get this to pass locally on my machine, any idea why? Running osx, example output:
`fh.read()`:
```
RUNSPEC
NOSIM
-- and more RUNSPEC
```
Expected:
```
RUNSPEC
-- and more RUNSPEC
```

Reply:
Have you tried running the sed command manually to see if you get the expected results? There are slight differences in how sed works on mac and linux, but I'm surprised if it's this big of a difference!

Reply:
Apparently:
> . -i works with GNU sed, recent versions of NetBSD, OpenBSD sed, but not OS X's sed. 

https://unix.stackexchange.com/a/131940

Reply:
Do we care about this? It's an copy of the current NOSIM job, so neither will work on OSX.

Reply:
Looks like the alternative is some funky rewrite, I support just keeping it as is.